python/erdos/streams.py

```diff
@@ -225,7 +233,8 @@ def concat(self, *other: "Stream") -> "OperatorStream":
         return streams_to_be_merged[0]
 
 
-class ReadStream:
+class ReadStream(Generic[A]):
```

Comment:
```suggestion
class ReadStream(Generic[T]):
```
I believe the scope of `TypeVar`s is restricted to the class, so the type variable `T` can be reused here.

Reply:
Please also annotate the `Message` types in both `read` and `try_read` below.